aea/protocols/base.py

```diff
@@ -54,7 +54,8 @@ def counterparty(self) -> Address:
 
         :return the address
         """
-        assert self._counterparty is not None, "Counterparty not set!"
+        if self._counterparty is None:
```

Comment:
That's not what I meant. We don't want to change None to something (empty string) when it is not set, please revert to previous implementation.

Reply:
You need to change the `__eq__` method. Rather than `and self.counterparty == other.counterparty` it should be `and self._counterparty == other._counterparty`

Reply:
Even though accessing the private attribute of `other` is not ideal, I think this is the cleanest way to compare here.